packages/nimble-components/src/rich-text/models/tests/markdown-serializer.spec.ts

```diff
@@ -22,7 +23,10 @@ describe('Markdown serializer', () => {
             OrderedList,
             ListItem,
             Bold,
-            Italic
+            Italic,
+            Link.extend({
```

Comment:
The tests are written to isolate the serializer from the component, yet they also require Tiptap's Node as an input to the `serializeDOMToMarkdown` method. However, this issue needs to be addressed, and in the same time we do not want to block this PR. We have created a tech debt issue, https://github.com/ni/nimble/issues/1533, and have started working on it to come up with a better testing strategy for the markdown-serializer. Please share your thoughts on the tech debt issue we've created.

> At this point can we just construct a `nimble-rich-text-editor` and use that with it's `setMarkdown` instead?

For links, these tests are already added in the describe block `asserting getMarkdown for rendered links` in the `nimble-rich-text-editor.spec.ts` file. We are also in the discussion of the right place to have these tests because constructing the actual `nimble-rich-editor` here is the same as writing tests in the component is what we thought.

> My concern is that if that's the case, what's the guarantee of stability of the HTML snippets captured in this file?

This is because we are using the `setContent` of Tiptap which accepts HTML strings in any form that can render as we wanted in the editor. For example, when I tried setting the `<em><strong>Bold</strong></em>` (which is not how it renders in editor DOM) through `setContent`, it renders as `<strong><em>Bold</em></strong>` based on the priority set in the Tiptap schema. 

> That's not a valid absolute link. What is this trying to even assert?

Yes, this should not be done in this way in the test files and I have updated all the text content used here to proper absolute links. The point we tried to test here is that the content of the `<a>` tag is just wrapped with the `<>` brackets when getting the markdown string to it due to the constraints we have with the prosemirror-markdown serializer. We added a comment about the rationale we went with that approach in `markdown-serializer.ts`.

To conclude, we will revisit and update the testing strategy for the serializer in a separate PR and we will consider all of the above and make sure to come up with the right possible way before we make major changes to the editor component.

Reply:
Sounds good. And a valid option may be that we find independently testing the markdown serializer to not be valuable. We never will pass HTML to it directly, only the TipTap editor will. So the test we want is that interactions with the tiptap editor produce expected markdown in the end. The HTML intermediate representation is a detail we don't really care about much as long as the same interactions in the editor produce the same markdown. But yes, continue discussion in the tech debt task.

Reply:
Sure, we will also see how we can test the anchors in Chromatic tests.